dask/dataframe/tests/test_groupby.py

```diff
@@ -1502,3 +1498,44 @@ def test_groupby_select_column_agg():
     actual = ddf.groupby('A')['B'].agg('var')
     expected = pdf.groupby('A')['B'].agg('var')
     assert_eq(actual, expected)
+
+
+@pytest.mark.skipif(PANDAS_VERSION < '0.23.0',
+                    reason="Need 0.23.0 for observed=True parameter")
+@pytest.mark.parametrize('df', [
+    pd.DataFrame({
+        'c1': pd.Categorical(['v11', 'v12', 'v11'], ['v11', 'v12', 'v13']),
+        'c2': pd.Categorical(['v21', 'v22', 'v23']),
+        'c3': [1, 4, 7]})])
+@pytest.mark.parametrize('cols_group, cols_out', [('c1', 'c3'),
+                                                  ('c1', ['c3']),
+                                                  (['c1', 'c2'], None),
+                                                  (['c1', 'c2'], 'c3'),
+                                                  (['c2', 'c1'], ['c3'])])
+def test_groupby_observed(df, agg_func, cols_group, cols_out):
+
+    # nunique is not supported in specs
+    if agg_func == 'nunique':
+        return
+
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    expected_g = df.groupby(cols_group, observed=True)
+    result_g = ddf.groupby(cols_group, observed=True)
+    if cols_out is not None:
+        expected_g = expected_g[cols_out]
+        result_g = result_g[cols_out]
+
+    expected_f = getattr(expected_g, agg_func)
+    result_f = getattr(result_g, agg_func)
+
+    expected = expected_f()
+    result = result_f().compute()
```

Comment:
Remove `.compute`. assert_eq will do that.

Reply:
I needed it; otherwise assert_eq fails. It boils down to dask not really supporting multiindex